kubera/authentication/create_user.go

```diff
@@ -0,0 +1,87 @@
+package authentication
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/mayadata-io/kubera-api-testing/kubera/connection"
+)
+
+// UserCreation contains structure of the input data to create NewUser
+type UserCreation struct {
+	FirstName  string         `json:"firstName"`
+	LastName   string         `json:"lastName"`
+	Credential UserCredential `json:"password"`
+}
+
+// UserCredential contains structure of the user secrets input
+type UserCredential struct {
+	Email    string `json:"publicValue"`
+	Password string `json:"secretValue"`
+}
+
+// String print UserCreation in pretty format
+func (t UserCreation) String() string {
+	if t.Credential.Password != "" {
+		t.Credential.Password = "XXXX"
+	}
+	raw, err := json.MarshalIndent(
+		t,
+		" ",
+		".",
+	)
+	if err != nil {
+		panic(err)
+	}
+	return string(raw)
+}
+
+// UserCreationConfig defines the structure of user input fields
+type UserCreationConfig struct {
+	UserCreation
+}
+
+// NewUserCreator returns a new instance of UserCreation
+func NewUserCreator(config UserCreationConfig) *UserCreation {
+	return &UserCreation{
+		FirstName: config.FirstName,
+		LastName:  config.LastName,
+		Credential: UserCredential{
+			Email:    config.Credential.Email,
+			Password: config.Credential.Password,
+		},
+	}
+}
+
+// Create is  a method , It creates new user in kubera by making -
+//  post reqest to API server
+func (t *UserCreation) Create() (Token, error) {
+	client := resty.New()
+	// url := connection.Get().HostName + "v3/localauthconfig/"
+	url := fmt.Sprintf("%s/v3/localauthconfig/", connection.Get().HostName)
```

Comment:
localauthconfig can be some field within UserCreation struct.
In other words, how do we test google auth, github auth & so on.
When are we going to test google/github based auths?

Reply:
Google and Github auth have totally different kind of body and endpoint .